cli/module_generate.go

```diff
@@ -347,19 +326,28 @@ func catchResolveOrgErr(cCtx *cli.Context, c *viamClient, newModule *moduleInput
 }
 
 // populateAdditionalInfo fills in additional info in newModule.
-func populateAdditionalInfo(newModule *moduleInputs) {
+func populateAdditionalInfo(newModule *common.ModuleInputs) {
 	newModule.GeneratedOn = time.Now().UTC()
 	newModule.GeneratorVersion = version
 	newModule.ResourceSubtype = strings.Split(newModule.Resource, " ")[0]
 	newModule.ResourceType = strings.Split(newModule.Resource, " ")[1]
 
 	titleCaser := cases.Title(language.Und)
-	replacer := strings.NewReplacer("_", "", "-", "")
-	newModule.ModulePascal = replacer.Replace(titleCaser.String(newModule.ModuleName))
+	replacer := strings.NewReplacer("_", " ", "-", " ")
+	spaceReplacer := strings.NewReplacer(" ", "", "_", "", "-", "")
+	newModule.ModulePascal = spaceReplacer.Replace(titleCaser.String(replacer.Replace(newModule.ModuleName)))
+	newModule.ModuleCamel = strings.ToLower(string(newModule.ModulePascal[0])) + newModule.ModulePascal[1:]
+	newModule.ModuleLowercase = strings.ToLower(newModule.ModulePascal)
 	newModule.API = fmt.Sprintf("rdk:%s:%s", newModule.ResourceType, newModule.ResourceSubtype)
-	newModule.ResourceSubtypePascal = replacer.Replace(titleCaser.String(newModule.ResourceSubtype))
-	newModule.ModelPascal = replacer.Replace(titleCaser.String(newModule.ModelName))
+	newModule.ResourceSubtypePascal = spaceReplacer.Replace(titleCaser.String(replacer.Replace(newModule.ResourceSubtype)))
+	if newModule.Language == golang {
+		newModule.ResourceSubtype = spaceReplacer.Replace(newModule.ResourceSubtype)
+	}
+	newModule.ResourceTypePascal = spaceReplacer.Replace(titleCaser.String(replacer.Replace(newModule.ResourceType)))
+	newModule.ModelPascal = spaceReplacer.Replace(titleCaser.String(replacer.Replace(newModule.ModelName)))
 	newModule.ModelTriple = fmt.Sprintf("%s:%s:%s", newModule.Namespace, newModule.ModuleName, newModule.ModelName)
+	newModule.ModelCamel = strings.ToLower(string(newModule.ModelPascal[0])) + newModule.ModelPascal[1:]
+	newModule.ModelLowercase = strings.ToLower(newModule.ModelPascal)
```

Comment:
(q) It's pretty hard for me to verify from looking at this that we're doing the right thing. Just to confirm, did you try using the generated module and make sure it compiled/ran successfully?

Reply:
yes I have generated a module for every component and service (both in go and python for comparison)! I wasn't sure how to provide evidence of this without posting screenshots of every resource